### PR TITLE
feat: Transfer support customize footer

### PR DESCRIPTION
**Pull request by xrkffgg**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/28082

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Transfer support custom left and right footer.          |
| 🇨🇳 Chinese | Transfer 新增自定义左右 footer。       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/transfer/demo/advanced.md [hunk 1]
```diff
@@ -50,11 +50,20 @@ class App extends React.Component {
     this.setState({ targetKeys });
   };
 
-  renderFooter = () => (
-    <Button size="small" style={{ float: 'right', margin: 5 }} onClick={this.getMock}>
-      reload
-    </Button>
-  );
+  renderFooter = (props, direction) => {
+    if (direction == 'left') {
```
Comment (xrkffgg) [inline components/transfer/demo/advanced.md:54]:
```suggestion
    if (direction === 'left') {
```

File: components/transfer/list.tsx [hunk 2]
```diff
@@ -319,10 +319,12 @@ export default class TransferList<
       showSelectAll,
       showRemove,
       pagination,
+      direction,
     } = this.props;
 
     // Custom Layout
-    const footerDom = footer && footer(this.props);
+    const footerDom =
+      footer && (footer.length < 2 ? footer(this.props) : footer(this.props, direction));
```
Comment (afc163) [inline components/transfer/list.tsx:327]: 这个 this.props 没啥用，把这个位置占住了，蛋疼。

Comment (afc163) [reply components/transfer/list.tsx:327]: 改成 `{ direction }` 吧，方便后续扩展。

File: components/transfer/list.tsx [hunk 3]
```diff
@@ -59,7 +59,10 @@ export interface TransferListProps<RecordType> extends TransferLocale {
   itemUnit: string;
   itemsUnit: string;
   renderList?: RenderListFunction<RecordType>;
-  footer?: (props: TransferListProps<RecordType>, direction?: TransferDirection) => React.ReactNode;
+  footer?: (
+    props: TransferListProps<RecordType>,
+    obj?: { direction: TransferDirection },
```
Comment (afc163) [inline components/transfer/list.tsx:64]:
```suggestion
    info?: { direction: TransferDirection },
```

**Review by afc163 — APPROVED** [review]

**Pull request merged**